relay-server/src/actors/processor.rs

```diff
@@ -1124,6 +1125,50 @@ impl EnvelopeProcessorService {
         });
     }
 
+    fn process_recording_timed(
+        &self,
+        bytes: &[u8],
+        limit: usize,
+    ) -> Result<Vec<u8>, recording::RecordingParseError> {
+        // Check for null byte condition.
+        if bytes.is_empty() {
+            return Err(recording::RecordingParseError::Message("no data found"));
+        }
+
+        let mut split = bytes.splitn(2, |b| b == &b'\n');
+        let header = split
+            .next()
+            .ok_or(recording::RecordingParseError::Message("no headers found"))?;
+
+        let body = match split.next() {
+            Some(b"") | None => {
+                return Err(recording::RecordingParseError::Message("no body found"))
+            }
+            Some(body) => body,
+        };
+
+        let buffer = metric!(timer(RelayTimers::ReplayRecordingProcessing), {
+            recording::decompress(body, limit)?
+        });
+        let mut events = metric!(timer(RelayTimers::ReplayRecordingProcessing), {
+            recording::deserialize(buffer)?
+        });
+
+        metric!(timer(RelayTimers::ReplayRecordingProcessing), {
+            recording::strip_pii(&mut events)
+                .map_err(recording::RecordingParseError::ProcessingAction)?
+        });
+
+        let bytes = metric!(timer(RelayTimers::ReplayRecordingProcessing), {
```

Comment:
do we have to create a unique timer for each thing we process to get the differentiated metrics? or does it use the module name or something?

Reply:
Oversight.  Nice catch.